tensorflow_graphics/projects/mesh_rcnn/branches/mesh_head.py

```diff
@@ -0,0 +1,190 @@
+# Copyright 2020 The TensorFlow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Mesh refinement branch of Mesh R-CNN"""
+
+import tensorflow as tf
+from tensorflow.keras import layers as keras_layers
+
+import tensorflow_graphics.geometry.convolution.utils as conv_utils
+from tensorflow_graphics.nn.layer.graph_convolution import \
+  FeatureSteeredConvolutionKerasLayer
+from tensorflow_graphics.projects.mesh_rcnn.ops.vertex_alignment import \
+  vert_align
+
+
+class MeshRefinementLayer(keras_layers.Layer):
+  """
+  Implements the Mesh refinement head of Mesh R-CNN.
+  """
+
+  def __init__(self,
+               n_stages,
+               n_gconvs,
+               gconv_dim,
+               gconv_init='normal',
+               name='MeshRefinementHead',
+               **kwargs):
+    """
+    Args:
+      n_stages: Number of refinement stages.
+      n_gconvs: Number of graph convolutions per stage.
+      gconv_dim: output dimensionality of graph convolutions.
+      gconv_init: String denoting how graph convolutions should be initialized.
+      name: An optional name for the layer.
+      **kwargs: Optional keyword arguments that get passed to the base layer.
+    """
+    super(MeshRefinementLayer, self).__init__(name=name, **kwargs)
+
+    self.n_stages = n_stages
+    self.n_gconvs = n_gconvs
+    self.gconv_dim = gconv_dim
+    self.gconv_init = gconv_init
+
+    self.stages = []
+
+  def build(self, input_shape):
+    for i in range(self.n_stages):
+      vertex_features_dim = 0 if i == 0 else self.gconv_dim
```

Comment:
Why is first layer 0-D? SHouldnt this be 3D? or 6D depending if you are using XYZ, or XYZ + NXNYNZ